Packageset membership (#11)
canonical/ubuntu-maintainers-handbook: MembershipInPackageSet.md

Membership in Package Set
-------------------------

> Upload rights can be given for certain package-sets, such as 'server' or 'desktop'.  It's original design intent was for individuals who will only be working on a very small set of packages.

Reviewer: It's -> Its

For this reason, some people skip this level and head straight for MOTU or core-dev instead; if you already have strong packaging experience via another distro you certainly can consider doing similarly.

That said, even if you intend to eventually apply for core-dev, gaining package-set first can be an effective way to build towards those roles, allowing you to upload your own work (within limits) and participate in reviews and sponsorship of co-workers.

Application Process
-------------------

> 0.  (Optional) If you have been involved in the Ubuntu community for a while in any fashion and you're not already a regular Ubuntu Member, add yourself to the next [Ubuntu Membership Board meeting](https://wiki.ubuntu.com/Membership/Boards), following its guidelines for what to include in your wiki page.

Reviewer: The general bar is "sustained and significant contributions" rather than "any fashion" (from https://wiki.ubuntu.com/Membership/NewMember). However if the sustained and significant contributions are in Ubuntu development, then [Contributing Developer](https://wiki.ubuntu.com/UbuntuDevelopers#ContribDev) is perhaps more appropriate since the bar is otherwise the same. This would be reviewed by the DMB rather than the general Ubuntu membership boards, and a successful applicant is implicitly granted Ubuntu Membership. So if recommending anything at all, I'd recommend this over seeking Ubuntu Membership for developers - unless this person has made "sustained and significant contributions" in non-developer areas in Ubuntu.

But it also seems like an extra unnecessary step to me to even suggest it, unless a contributor intends to stop there, or wants progressive recognition. I can only recall one occasion where it was used in the past few years, and that was an exceptional case. Other than that, I don't think we've recommended this path before, unless it was before my time (pre 2011).

Ideally, you should have a solid mastery of the [basic packaging skills](https://packaging.ubuntu.com/html/) for Debian/Ubuntu distributions, including the following:

  * [Fixing bugs in packages](PackageFixing)
  * Building binary packages from source using sbuild or debuild in a chroot or lxc environment
  * Creating the initial packaging for new software
  * Merging updates from Debian
  * Backporting patches
  * Forwarding bugs and patches to Debian and to upstream maintainers
  * Stable Release Update process

Make sure you've done each of the above items at least a couple of times.  If you're not comfortable with any of these, plan on doing some additional practice with them.

You should also work towards understanding some more advanced packaging topics:

  * Purpose of the [different files in debian/](https://packaging.ubuntu.com/html/debian-dir-overview.html)
  * [Debian policy](http://www.debian.org/doc/debian-policy/)
  * [Ubuntu's release process](https://wiki.ubuntu.com/UbuntuDevelopment/ReleaseProcess), including the
    [freeze exception process](https://wiki.ubuntu.com/FreezeExceptionProcess)
  * Running [Autopkgtest](PackageTests)

>   * Troubleshooting migration of packages from -proposed

Reviewer: Link: https://wiki.ubuntu.com/ProposedMigration

Collaboration with the upstream(s) related to the software within your packageset is important as well.  Gain experience with your upstreams' bug reporting processes, patch review processes, and testing/CI systems if you haven't so far.  In general, the narrower the number of packages you plan to focus on, the stronger your collaboration with upstream you should be able to show.

And on the flip side, if you are a Canonical employee you will need to be able to negotiate the distinction between Ubuntu governance and Canonical priorities, within your area of focus.  Sometimes these can appear to collide and require a certain level of diplomacy to find solutions that work well for both sides.  The less experience you have in Debian or other open source communities, the more thought you'll want to put into this.

Finally, you'll know you're past ready for applying if anyone ever asks, "How do you not already have upload rights??"

Further Information
-------------------

> https://wiki.ubuntu.com/RobieBasak/DMB/CoreDev

Reviewer 2: This line needs two spaces in the end for Markdown to render it correctly.
```suggestion
https://wiki.ubuntu.com/RobieBasak/DMB/CoreDev  
```

https://wiki.ubuntu.com/UbuntuDevelopers

> Once you've been granted packageset upload permissions, there's at least three distinct directions you could embark on, depending on your goals:

Reviewer 2: ```suggestion
Once you've been granted packageset upload permissions, there are at least three distinct directions you could embark on, depending on your goals:
```

>   * MOTU, if you want to get depth into the Ubuntu packaging world.

Reviewer 2: Should be ideal to put in a link for those who don't know MOTU (yet).
```suggestion
  * [MOTU](https://wiki.ubuntu.com/MOTU), if you want to get depth into the Ubuntu packaging world.
```

Ideally, you should have a solid mastery of the [basic packaging skills](https://packaging.ubuntu.com/html/) for Debian/Ubuntu distributions, including the following:

  * [Fixing bugs in packages](PackageFixing)
  * Building binary packages from source using sbuild or debuild in a chroot or lxc environment
  * Creating the initial packaging for new software
  * Merging updates from Debian
  * Backporting patches
  * Forwarding bugs and patches to Debian and to upstream maintainers
  * Stable Release Update process

Make sure you've done each of the above items at least a couple of times.  If you're not comfortable with any of these, plan on doing some additional practice with them.

You should also work towards understanding some more advanced packaging topics:

  * Purpose of the [different files in debian/](https://packaging.ubuntu.com/html/debian-dir-overview.html)
  * [Debian policy](http://www.debian.org/doc/debian-policy/)
  * [Ubuntu's release process](https://wiki.ubuntu.com/UbuntuDevelopment/ReleaseProcess), including the
    [freeze exception process](https://wiki.ubuntu.com/FreezeExceptionProcess)
  * Running [Autopkgtest](PackageTests)

>   * Troubleshooting [https://wiki.ubuntu.com/ProposedMigration](migration of packages) from -proposed

Reviewer 3: IMHO this (the "you don't need to be an expert, but understand the concepts") list needs to have a few more added:
- library transitions https://people.canonical.com/~ubuntu-archive/transitions/
- Seed management https://wiki.ubuntu.com/SeedManagement & https://people.canonical.com/~ubuntu-archive/component-mismatches-proposed.html
- NBS https://people.canonical.com/~ubuntu-archive/nbs.html
- Multiarch https://wiki.ubuntu.com/MultiarchCross

Those are in order of importance and e.g. the top two came up in my CoreDev application.
